semgrep-core/src/synthesizing/Unit_synthesizer_targets.ml

```diff
@@ -0,0 +1,114 @@
+(*s: semgrep/matching/Unit_matcher.ml *)
+open Common
+open OUnit
+module R = Mini_rule
+
+let test_path = "../../../tests/OTHER/synthesizing/targets/"
+
+(* Format: file, list of target ranges, expected pattern line. *)
+let stmt_tests =
+  [
+    (* Motivating example. *)
+    ("hello.py", [ "2:0-2:5"; "5:0-5:8" ], "$X(3, ...)");
```

Comment:
An alternative would be to put the content directly. It would then be easier to see
the input / expected output relation.
That would require to calls functions from semgrep-core/parsing/ to correctly parse the code into generic ASTs.